### DISCUSSION TITLE
SetDecSep, does it work?

### DISCUSSION THREAD

**Discussion by ramanamit** [beltoforion/muparser#135] [General]
Hi Guys,

Does SetDecSep work?

**Comment by beltoforion** [comment]
Its supposed to.

**Comment by ramanamit** [comment]
Maybe I am not using it in a right way. Please correct me if I am wrong.
I made some changes to example code. This how it looks like:

```
#include <iostream>
#include "muParser.h"

double MySqr(double a_fVal) {  return a_fVal*a_fVal; }

int main(int argc, char *argv[])
{
	try
	{
		double var_a = 1;
		mu::Parser p;
		p.SetDecSep(',');
		p.SetArgSep(';');
		p.SetThousandsSep('.');

		p.DefineVar("a", &var_a); 
		p.DefineFun("MySqr", MySqr); 
		p.SetExpr("MySqr(a)*3,5");

		for (std::size_t a = 0; a < 6; ++a)
		{
			var_a = a;  // Change value of variable a
			std::cout << p.Eval() << std::endl;
		}
	}
	catch (mu::Parser::exception_type &e)
	{
		std::cout << e.GetMsg() << std::endl;
	}
	
	return 0;
}
```

The result is:
**Unexpected token ",5 " found at position 10.**

SetArgSep function seems to work fine. It stores a given value in a ParserTokenReader's field: m_cArgSep.
But what about SetDecSep function? It does not look like it is stored anywhere.

In some parts of the world it is common to use comma as decimal separator.

**Comment by beltoforion** [comment]
Yeah, it seems broken now. I created a ticket.
